gui/wxpython/location_wizard/wizard.py

```diff
@@ -1447,80 +1444,115 @@ class WKTPage(TitledPage):
         self.sizer.SetCols(5)
         self.sizer.SetRows(8)
         
-        self.wktfile = ''
+        self.customstring = ''
+        self.parent = parent
 
-        # create controls
-        self.lfile = self.MakeLabel(_("WKT .prj file:"))
-        self.tfile = self.MakeTextCtrl(size=(300, -1))
-        self.bbrowse = self.MakeButton(_("Browse"))
+        # widgets
+        self.text_wkt = self.MakeTextCtrl(size=(400, 200),
+                                                  style=wx.TE_MULTILINE)
+        self.label_wkt = self.MakeLabel(
+            _("Enter WKT parameters string:"))
 
-        # do layout
-        self.sizer.Add(self.lfile, flag=wx.ALIGN_LEFT |
-                       wx.ALIGN_CENTRE_VERTICAL |
-                       wx.ALL, border=5, pos=(1, 1))
-        self.sizer.Add(self.tfile, flag=wx.ALIGN_LEFT |
-                       wx.ALIGN_CENTRE_VERTICAL |
-                       wx.ALL, border=5, pos=(1, 2))
-        self.sizer.Add(self.bbrowse, flag=wx.ALIGN_LEFT |
-                       wx.ALL, border=5, pos=(1, 3))
-        self.sizer.AddGrowableCol(3)
+        # layout
+        self.sizer.Add(self.label_wkt,
+                       flag=wx.ALIGN_LEFT | wx.ALL,
+                       border=5, pos=(1, 1))
+        self.sizer.Add(self.text_wkt,
+                       flag=wx.ALIGN_LEFT | wx.ALL | wx.EXPAND,
+                       border=5, pos=(2, 1), span=(1, 2))
+        self.sizer.AddGrowableRow(2)
+        self.sizer.AddGrowableCol(2)
 
-        self.bbrowse.Bind(wx.EVT_BUTTON, self.OnBrowse)
-        self.tfile.Bind(wx.EVT_TEXT, self.OnText)
+        self.text_wkt.Bind(wx.EVT_TEXT, self.GetWktstring)
         self.Bind(wiz.EVT_WIZARD_PAGE_CHANGING, self.OnPageChanging)
         self.Bind(wiz.EVT_WIZARD_PAGE_CHANGED, self.OnEnterPage)
 
     def OnEnterPage(self, event):
-        if len(self.wktfile) == 0:
+        if len(self.customstring) == 0:
             # disable 'next' button by default
             wx.FindWindowById(wx.ID_FORWARD).Enable(False)
         else:
             wx.FindWindowById(wx.ID_FORWARD).Enable(True)
 
         event.Skip()
-
+        
     def OnPageChanging(self, event):
-        if event.GetDirection() and not os.path.isfile(self.wktfile):
-            event.Veto()
-        self.GetNext().SetPrev(self)
+        if event.GetDirection():
+            self.custom_dtrans_string = ''
 
-        event.Skip()
+            if self.customstring.find('+datum=') < 0:
+                self.GetNext().SetPrev(self)
+                return
 
-    def OnText(self, event):
-        """File changed"""
-        self.wktfile = event.GetString()
-        nextButton = wx.FindWindowById(wx.ID_FORWARD)
-        if len(self.wktfile) > 0 and os.path.isfile(self.wktfile):
-            if not nextButton.IsEnabled():
-                nextButton.Enable(True)
-        else:
-            if nextButton.IsEnabled():
-                nextButton.Enable(False)
+            # check for datum tranforms
```

Comment:
To keep things consistent with the rest, the WKTPage doesn't change much (except for the layout). g.proj is called in Summary page:

```
            elif coordsys == 'wkt':
                ret, projlabel, err = RunCommand(
                    'g.proj', flags='jft', wkt=self.parent.wktpage.wktfile, **extra_opts)
```

so you need to adjust the code here. So OnPageChanging should stay similar to what was there before. The RunCommand in Summary page should look something like this:

```
            elif coordsys == 'wkt':
                ret, projlabel, err = RunCommand(
                    'g.proj', flags='jft', wkt="-", stdin=self.parent.wktpage.wktstring, **extra_opts)
```
(not tested)